specs/MultiProfile.md

```diff
@@ -0,0 +1,240 @@
+                            .query<ICoreWebView2Controller>();
+                    return OnCreateCoreWebView2ControllerCompleted(result, controller.get());
+                })
+                .Get()));
+    }
+    else
+    {
+        CHECK_FAILURE(webViewEnvironment4->CreateCoreWebView2ControllerWithOptions(
+            m_mainWindow, options.Get(),
+            Callback<ICoreWebView2CreateCoreWebView2ControllerCompletedHandler>(
+                this, &AppWindow::OnCreateCoreWebView2ControllerCompleted)
+                .Get()));
+    }
+
+    return S_OK;
+}
+```
+
+## Access the profile property of WebView2
+
+### Win32 C++
+
+```cpp
+HRESULT AppWindow::OnCreateCoreWebView2ControllerCompleted(HRESULT result, ICoreWebView2Controller* controller)
+{
+    // ...
+
+    m_controller = controller;
+    
+    // Gets the webview object from controller.
+    wil::com_ptr<ICoreWebView2> coreWebView2;
+    CHECK_FAILURE(m_controller->get_CoreWebView2(&coreWebView2));
+    coreWebView2.query_to(&m_webView);
+    m_webView3 = coreWebView2.try_query<ICoreWebView2_3>();
+    
+    // Gets the profile property of webview.
+    wil::com_ptr<ICoreWebView2Profile> profile;
+    CHECK_FAILURE(m_webView3->get_Profile(&profile));
+    
+    // Accesses the profile object.
+    wil::unique_cotaskmem_string name;
+    CHECK_FAILURE(profile->get_ProfileName(&name));
+    BOOL inPrivateModeEnabled;
+    CHECK_FAILURE(profile->get_InPrivateModeEnablede(&inPrivateModeEnabled));
+    wil::unique_cotaskmem_string path;
+    CHECK_FAILURE(profile->get_ProfilePath(&path));
+    
+    // ...
+}
+```
+
+# API Details
+
+## Win32 C++
+
+```IDL
+interface ICoreWebView2ControllerOptions;
+interface ICoreWebView2Environment5;
+interface ICoreWebView2_7;
+interface ICoreWebView2Profile;
+
+[uuid(C2669A3A-03A9-45E9-97EA-03CD55E5DC03), object, pointer_default(unique)]
+interface ICoreWebView2ControllerOptions : IUnknown {
+  /// `ProfileName` property is to specify a profile name, which is only allowed to contain
+  /// the following ASCII characters with the maximum length as 64 and will be treated in a
+  /// case insensitive way.
+  ///    alphabet characters: a-z and A-Z
+  ///    digit characters: 0-9
+  ///    and '#', '@', '$', '(', ')', '+', '-', '_', '~', '.', ' ' (space).
+  /// Note: the text must not end with a period '.' or ' ' (space). And, although upper case letters are
+  /// allowed, they're treated just as lower case couterparts because the profile name will be mapped to
+  /// the real profile directory path on disk and Windows file system handles path names in a case-insensitive way.
+  [propget] HRESULT ProfileName([out, retval] LPWSTR* value);
+  /// Sets the `ProfileName` property.
+  [propput] HRESULT ProfileName([in] LPCWSTR value);
+
+  /// `InPrivateModeEnabled` property is to enable/disable InPrivate mode.
+  [propget] HRESULT InPrivateModeEnabled([out, retval] BOOL* enabled);
```

Comment:
To make this consistent with our other Is*Enabled properties, let's add 'Is' to the beginning of the name.

Also, we try to use the parameter name `value` for properties.

```suggestion
  [propget] HRESULT IsInPrivateModeEnabled([out, retval] BOOL* value);
```